src/vs/workbench/contrib/files/browser/explorerService.ts

```diff
@@ -358,6 +372,25 @@ export class ExplorerService implements IExplorerService {
 		}
 	}
 
+	// Reveal excludes
+	private matchesUpToRoot(item: ExplorerItem | undefined): boolean {
```

Comment:
Couple notes: 
- the name of this isn't very discriptive `shouldAutoRevealItem` might be better?
- walking up the directly tree for every item isn't great, the glob matcher should be able to know that `node_modules/**` matches `node_modules/abc/d` without stepping through every directory?

Reply:
I will change the name as you suggested.

The glob matcher can match files if specified, e.g. `**/node_modules/*` would work fine without recursive checking.
The reason for this implementation is that the settings are inherited from the `files.exclude` setting, which allows specifying both folder and file patterns for showing files in the explorer. That setting is passed recursively and only needs to be called once for every tree item.
However, I also don't think the performance impact is too serious, as:

1. Focusing on a file is not a super frequent operation
2. File structures do not tend to be super deep, in the case of `**/node_modules` it would be 4-5 match operations for a source file inside 

In that case, there are two options I would suggest here:

1.  Keep the matching as is, but only check for sibling clauses with the initial item, and do not pass as sibling clause for the parent matching.
2. Remove the recursive matching, and edit the default setting and setting description to say that you must explicitly specify file patterns (`**/node_modules/*`) and not folder patterns (`**/node_modules`)

Reply:
@jzyrobert Did we settle on anything here I think we only support file patterns for files.exclude so this should mirror that.